### PR TITLE
[mesh] option to add mesh layer from data source manager dialog

### PR DESCRIPTION
**Pull request by PeterPetrik**
add_mesh2

Note: I am sorry for the icon, take it as a placeholder :)

### CODE REVIEW

**Comment by nirvn** [comment]
Nice. Out of curiosity, is there a plan to also support adding mesh datasets via the browser panel?

**Comment by PeterPetrik** [comment]
@nirvn That is already implemented for 2dm files :)

File: src/app/qgisapp.h [hunk 1]
```diff
@@ -922,6 +922,12 @@ class APP_EXPORT QgisApp : public QMainWindow, private Ui::MainWindow
       */
     bool addRasterLayers( const QStringList &layerQStringList, bool guiWarning = true );
 
+    /**
+     * Adds a mesh layer directly without prompting user for location
+     * \returns true if successfully added layer
+     */
+    bool addMeshLayer( const QString &uri, const QString &providerKey );
```
Comment (nyalldawson) [inline src/app/qgisapp.h]: Should this be exposed through iface too? (Just like the raster/vector ones are)

**Comment by nirvn** [comment]
@PeterPetrik , nice -- can't wait for GRIB files to be supported too. Keep the great work coming in :)

**Comment by PeterPetrik** [comment]
@nirvn here you go https://github.com/qgis/QGIS/pull/7409

**Pull request merged**